state-chain/runtime/src/lib.rs

```diff
@@ -1260,8 +1260,19 @@ impl_runtime_apis! {
 		///
 		/// Note: This function must only be called through RPC, because RPC has its own storage buffer
 		/// layer and would not affect on-chain storage.
-		fn cf_pool_simulate_swap(from: Asset, to:Asset, amount: AssetAmount) -> Result<SwapOutput, DispatchErrorWithMessage> {
-			LiquidityPools::swap_with_network_fee(from, to, amount).map_err(Into::into)
+		fn cf_pool_simulate_swap(
+			from: Asset,
+			to: Asset,
+			amount: AssetAmount,
+			first_leg_additional_limit_orders: Vec<(Tick, U256)>,
+			second_leg_additional_limit_orders: Vec<(Tick, U256)>,
```

Comment:
not super happy with the verbosity, could wrap it up into a `SwapInput` struct but didn't feel like it would be much better